packages/xo-server-sdn-controller/src/index.js

```diff
@@ -304,15 +370,68 @@ class SDNController extends EventEmitter {
     )
 
     const center = await this._electNewCenter(privateNetwork, false)
-    this._poolNetworks.push({
+    const poolNetwork = {
       pool: pool.$ref,
       network: privateNetwork.$ref,
       starCenter: center?.$ref,
-    })
+    }
+    this._poolNetworks.push(poolNetwork)
     this._networks.set(privateNetwork.$id, privateNetwork.$ref)
     if (center !== undefined) {
       this._starCenters.set(center.$id, center.$ref)
     }
+
+    return poolNetwork
+  }
+
+  async _createCrossPoolPrivateNetwork({
+    xoPoolIds,
+    networkName,
+    networkDescription,
+    encapsulation,
+    xoPifIds,
+  }) {
+    const uuid = uuidv4()
+    const crossPoolNetwork = {
+      pools: [],
+      networks: [],
+      uuid,
+    }
+
+    log.debug('New cross-pool network created', { uuid })
+
+    for (const xoPoolId of xoPoolIds) {
+      const xoPool = this._xo.getObject(xoPoolId, 'pool')
+      const pool = this._xo.getXapiObject(xoPool)
+
+      const xoPifId = find(xoPifIds, id => {
+        const pif = this._xo.getXapiObject(this._xo.getObject(id, 'PIF'))
+        return pif.$pool.$ref === pool.$ref
+      })
+      const xoPif = this._xo.getObject(xoPifId, 'PIF')
+
+      const poolNetwork = await this._createPrivateNetwork({
+        xoPool,
+        networkName,
+        networkDescription,
+        encapsulation,
+        xoPif,
+      })
+
+      const network = pool.$xapi.getObjectByRef(poolNetwork.network)
+      await network.update_other_config('cross_pool_network_uuid', uuid)
+
+      crossPoolNetwork.pools.push(poolNetwork.pool)
+      crossPoolNetwork.networks.push(poolNetwork.network)
+      log.debug('Pool network added to cross-pool network', {
+        network: networkName,
+        pool: pool.name_label,
```

Comment:
If this is a cross-pool network, why is there only one pool?
I think I might know the answer to this, but I'd like to be sure :-)

Reply:
It's in a `for` loop. The pools are added 1 by 1.